include/zephyr/bluetooth/hci.h

```diff
@@ -32,7 +32,11 @@ extern "C" {
  *
  * @return The string representation of the HCI error code.
  */
+#ifdef CONFIG_BT_HCI_ERR_TO_STR
 const char *bt_hci_err_to_str(uint8_t hci_err);
+#else
+static inline const char *bt_hci_err_to_str(uint8_t hci_err) { return ""; }
+#endif
```

Comment:
Since it's now guarded, we should probably add a note in the documentation for it:

```
 * If @kconfig{CONFIG_BT_HCI_ERR_TO_STR} is not enabled, this just returns the empty string
```

Or something like that